### PR TITLE
Add ability to format links to twitter.com accounts

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
See the discussion on #3888 and #1849 — this attempts to clean up the stream UI because a LOT of people do use the crossposter which makes the stream presently almost unreadable when multiple accounts are mentioned.

### CODE REVIEW

**Comment by Gargron** [comment]
I agree that it's a real problem and this might be a practical solution. But I can't help but feel that we're giving up something with this move. That goddamn crossposter is a curse.

**Comment by ThisIsMissEm** [comment]
Yeah, crossposting is problematic, but that's currently how people are using the service, so let's pave the cowpath.

File: app/lib/formatter.rb [hunk 1]
```diff
@@ -208,6 +209,10 @@ def link_to_hashtag(entity)
     hashtag_html(entity[:hashtag])
   end
 
+  def link_to_twitter(username)
+    "<span class=\"h-card\"><a href=\"https://twitter.com/#{username}\" target=\"blank\" rel=\"noopener noreferrer\" class=\"u-url mention\"><i class=\"fa fa-twitter\"></i>&nbsp;@<span>#{username}</span></a></span>"
```
[the remark this answers is not in the document]
Comment (ThisIsMissEm) [reply app/lib/formatter.rb]: Okay, that's a fair compromise. I'll update to just make it a @<username>@twitter.com link

**Comment by Gargron** [comment]
I'm having second thoughts about this. What kind of message does this send? That it's okay to give special treatment to Twitter? That we're never getting rid of it? It seems antitethical to our ideals.

A lot of new people are using these Twitter to Mastodon cross-posters. But do we really want to encourage that? What it enables is "set and forget", where people set the cross-posting up and never visit Mastodon again. They won't respond to toots then. So it becomes fake activity, and waste of disk space and processing power, and spam on the public timelines.

Tumblr doesn't allow cross-posting *from* Twitter, only *to*, and the same was true for Postereus and other richer platforms. Cross-posting is marketing, and I now realize more than ever that Twitter to Mastodon cross-posting is hurting Mastodon.

**Comment by ThisIsMissEm** [comment]
I’d agree, but whilst crossposting exists, it feels like we should at least attempt to set it up in a way that’s not awful.

We can also look into “In case you missed it” type emails if people have unread notifications after a certain period of time.

i.e., make it easy to re-engage users, and bring them back to active on mastodon.

Just like embedding tweets (and other content) in mastodon would allow people to move conversations to mastodon. Likewise, we could improve the bridges to be better — I feel like they’re always going to exist for the forseeable future until people move across. It’d be great if the bridge emailed you to say “hey, we’ve found more people you know!” — at present you have to check back to find out who else has joined.

Whilst I understand the hesitation of adding twitter API integration into mastodon, it’d act as the “one true way”, allowing better control over the integration.

**Comment by Cassolotl** [comment]
I asked for this feature before any user-friendly crossposters existed, because sometimes I wanted to refer to someone on Twitter with `@username@instance` and having it actually link up made a lot of sense to me. Now that I use a crossposter, most of my original toots are crossposted tweets.

I understand that the introduction of usable crossposters changes the landscape quite a bit, and I'm wondering if perhaps this is something to be brought up with the crossposter developers. Perhaps some kind of... what is that thing called, a dead man's switch? If someone doesn't log into or post on Mastodon for x amount of time, the crossposter stops working. So I'll go and suggest this on Mastodon and see what happens.

**Comment by ThisIsMissEm** [comment]
@Gargron So, the consensus was to close this? I had seen cases where the @mistressemelia part of @mistressemelia@twitter.com was linked. I think this is a bug in Amaroq.

**Pull request closed without merging**